mlflow/pyspark/ml/__init__.py

```diff
@@ -158,7 +158,19 @@ def _traverse_stage(stage):
 
     yield stage
     if isinstance(stage, Pipeline):
-        for stage in stage.getStages():
+        original_sub_stages = stage.getStages()
+
+        try:
+            iter(original_sub_stages)
+            is_iterable = True
+        except TypeError:
+            is_iterable = False
```

Comment:
```suggestion
        try:
            iter(original_sub_stages)
        except TypeError:
            raise TypeError(...)
```

I think we can throw an error here.